src/shared/config/AuthXmlConfig.php

```diff
@@ -0,0 +1,35 @@
+<?php declare(strict_types = 1);
+namespace PharIo\Phive;
+
+class AuthXmlConfig implements AuthConfig {
+    /** @var XmlFile */
+    private $xmlFile;
+
+    /**
+     * AuthXmlConfig constructor.
+     */
+    public function __construct(XmlFile $xmlFile) {
+        $this->xmlFile = $xmlFile;
+    }
+
+    public function hasAuthentication(string $domain): bool {
+        $query  = \sprintf('//phive:domain[@host="%s"]', $domain);
+        $result = $this->xmlFile->query($query);
+
+        return $result->count() > 0;
+    }
+
+    public function getAuthentication(string $domain): Authentication {
+        $query  = \sprintf('//phive:domain[@host="%s"]', $domain);
+        $result = $this->xmlFile->query($query);
+
+        /** @var \DOMElement $auth */
+        $auth = $result->item(0);
+
```

Comment:
This operates on the assumption that the config has a node for this domain.
We should catch the case it's not. Trusting that `hasAuthentication` was called before is dangerous...

Reply:
On top of that: Is the name case-sensitive?

Reply:
I've done it correctly in `Environment`, but not here... I'll call `hasAuthentication` and throw an exception if needed